monkey/monkey_island/cc/ui/src/components/ui-components/AdvancedMultiSelect.js

```diff
@@ -41,81 +32,168 @@ function getObjectFromRegistryByRef(refString, registry) {
   return resolveObjectPath(refArray, registry);
 }
 
-function getFullDefinitionByKey(refString, registry, itemKey) {
-  let fullArray = getFullDefinitionsFromRegistry(refString, registry);
-  return fullArray.filter(e => (e.enum[0] === itemKey))[0];
-}
-
-function setPaneInfo(refString, registry, itemKey, setPaneInfoFnc) {
-  let definitionObj = getFullDefinitionByKey(refString, registry, itemKey);
-  setPaneInfoFnc({title: definitionObj.title, content: definitionObj.info, link: definitionObj.link});
-}
-
 function getDefaultPaneParams(refString, registry) {
   let configSection = getObjectFromRegistryByRef(refString, registry);
   return ({title: configSection.title, content: configSection.description});
 }
 
-function AdvancedMultiSelect(props) {
-  const [masterCheckbox, setMasterCheckbox] = useState(true);
-  const {
-    schema,
-    id,
-    options,
-    value,
-    required,
-    disabled,
-    readonly,
-    multiple,
-    autofocus,
-    onChange,
-    registry
-  } = props;
-  const {enumOptions} = options;
-  const [infoPaneParams, setInfoPaneParams] = useState(getDefaultPaneParams(schema.items.$ref, registry));
-  getDefaultPaneParams(schema.items.$ref, registry);
-  const selectValue = cloneDeep(value);
-  return (
-    <div className={'advanced-multi-select'}>
-      <Card.Header>
-        <Button key={`${props.schema.title}-button`} value={value}
+class AdvancedMultiSelect extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      masterCheckboxState: this.getMasterCheckboxState(props.value),
+      infoPaneParams: getDefaultPaneParams(props.schema.items.$ref, props.registry)
+    };
+
+    this.onMasterCheckboxClick = this.onMasterCheckboxClick.bind(this);
+    this.onChildCheckboxClick = this.onChildCheckboxClick.bind(this);
+    this.setPaneInfo = this.setPaneInfo.bind(this, props.schema.items.$ref, props.registry);
+  }
+
+  onMasterCheckboxClick() {
+    let newValues = this.props.options.enumOptions.map(({value}) => value);
```

Comment:
If you returned variable initialization to the top from `render` method, you could use `enumOptions.map` instead of `this.props.options.enumOptions.map`. Why did you move it to `render()` btw?

Reply:
What does `({value}) => value` do?

Reply:
On master checkbox click, we never want to enable everything, that's why previous method had declaration: `function onMasterCheckboxClick(checkboxValue, **defaultArray**, onChangeFnc) {`. Either revert to selecting default values or make this behavior changeable, but leave "select only default values" behavior as default.

Reply:
This method was more readable with it's previous structure of "set next value based on master checkbox state". Now it's more like "set value to something (full array probably), but then change it if master checkbox has one specific state". How I would solve it is 
```
if (this.state.masterCheckboxState == MasterCheckboxState.ALL) { 
    let newValues = [];
} else {
    let newValues = defaultArray;
}
```

Reply:
> Why did you move it to render() btw?

I didn't so much move it to `render()` as I just left it where it was and transformed the component from a function to a class. 

> If you returned variable initialization to the top from render method, you could use enumOptions.map instead of this.props.options.enumOptions.map

This is probably worth doing.

> What does `({value}) => value` do?

`enumOptions` is an array of objects that look like `{label: "LABEL", value: "VALUE"}`. Using map with `({value}) => value` returns an array containing only the value strings.

> On master checkbox click, we never want to enable everything

I think having a master checkbox that behaves differently than every other master checkbox users have ever encountered is a recipe for frustration. If the goal of the control is to make it easy for the user to select/deselect all options, a master checkbox makes sense, but it should behave like a master checkbox. If the goal is to make it easy for the user to reset to the default options, a button that says "reset to default" or something similar is a much better way to communicate to the user what the control is supposed to do. Please re-read the PR description where I discuss the change in behavior. 

> This method was more readable...

Good feedback. I went back and forth on this. I'll change it back.